packages/flutter/test/widgets/framework_test.dart

```diff
@@ -1619,6 +1619,32 @@ void main() {
     await pumpWidget(Container());
     expect(states, <String>['deactivate', 'dispose']);
   });
+
+  testWidgets('Getting the render object of an unmounted element throws', (WidgetTester tester) async {
+    await tester.pumpWidget(const _StatefulLeaf());
+    final StatefulElement element = tester.element<StatefulElement>(find.byType(_StatefulLeaf));
+    expect(element.state, isA<State<_StatefulLeaf>>());
+    expect(element.widget, isA<_StatefulLeaf>());
+    // Replace the widget tree to unmount the element.
+    await tester.pumpWidget(Container());
+
+    try {
+      element.findRenderObject();
+      fail('Expected exception');
+    } catch (e) {
+      // Make sure we only throw this type of error and no other.
+      final FlutterError error = e as FlutterError;
```

Comment:
```suggestion
    } on FlutterError catch (e) {
```

Reply:
Ahh I had it this way and then mistakenly worried we'd somehow miss some other type of exception.